src/Entities/Player.cpp

```diff
@@ -1447,13 +1447,16 @@ void cPlayer::TeleportToCoords(double a_PosX, double a_PosY, double a_PosZ)
 	//  ask plugins to allow teleport to the new position.
 	if (!cRoot::Get()->GetPluginManager()->CallHookEntityTeleport(*this, m_LastPosition, Vector3d(a_PosX, a_PosY, a_PosZ)))
 	{
-		SetPosition(a_PosX, a_PosY, a_PosZ);
-		FreezeInternal(GetPosition(), false);
-		m_LastGroundHeight = a_PosY;
-		m_bIsTeleporting = true;
-
-		m_World->BroadcastTeleportEntity(*this, GetClientHandle());
-		m_ClientHandle->SendPlayerMoveLook();
+		m_World->QueueTask([this, a_PosX, a_PosY, a_PosZ](cWorld & a_World)
```

Comment:
No, this is not good. If the player disconnects between the enqueue and the execution of the task, the server will crash. You cannot capture `this`.